### PR TITLE
feat: add assert() operator

### PR DESCRIPTION
**Pull request by zzlk**

### CODE REVIEW

**Comment by zzlk** [comment]
hmm, panics don't unwind in wasm so need to disable these tests for wasm

File: hydroflow_lang/src/graph/ops/assert.rs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+/// The input stream will be collected into a vector and then compared with the input argument. If there is a difference then the program will panic.
+///
+/// assert() is mainly useful for testing and documenting the behavior of hydroflow code inline.
+///
+/// assert() will pass through the items to its output channel unchanged and in-order.
+///
+/// ```hydroflow
+/// source_iter([1, 2, 3])
+///     -> assert([1, 2, 3])
+///     -> assert([1, 2, 3]);
```
Comment (MingweiSamuel) [inline hydroflow_lang/src/graph/ops/assert.rs:21]: maybe make the last operator for_each println

File: hydroflow_lang/src/graph/ops/assert.rs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+               _| {
+        let vec = &arguments[0];
+
+        let (write_prologue, write_iterator, write_iterator_after) = if is_pull {
+            let input = &inputs[0];
+
+            (
+                Default::default(),
+                quote_spanned! {op_span=>
+                    let input_vec = #input.collect::<::std::vec::Vec<_>>();
```
Comment (MingweiSamuel) [inline hydroflow_lang/src/graph/ops/assert.rs:63]: should make a separate ident for `input_vec`, like

```
let vec_ident = wc.make_ident(format!("vec_{}", side));
```

**Review by MingweiSamuel — APPROVED** [review]
Looks good, add a note about how it behaves with ticks

File: hydroflow_lang/src/graph/ops/assert.rs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+                    assert_eq!(input_vec, #vec, "lhs = dataflow input, rhs = expected value (provided as argument to assert())");
```
[the remark this answers is not in the document]
Comment (zzlk) [reply hydroflow_lang/src/graph/ops/assert.rs:65]: Yeah, I think that is the simplest thing to implement for now, later we can think about how to assert stuff across ticks but I'm not sure what that would look like.

File: hydroflow_lang/src/graph/ops/assert.rs [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+                                    accum.push(v);
+                                    #context.state_ref(#assert_data_ident).set(accum);
+                                });
+                            }
+                        }
+                        [output] => {
+                            quote_spanned! {op_span=>
+                                let #ident = #root::pusherator::inspect::Inspect::new(|v| {
+                                    let mut accum = #context.state_ref(#assert_data_ident).take();
+                                    accum.push(v);
```
Comment (zzlk) [inline hydroflow_lang/src/graph/ops/assert.rs:97]: uh how does this even work? would v be &T..?

**Pull request merged**